docs/mrd_yardl.md

```diff
@@ -0,0 +1,50 @@
+(FileFormat)=
+# MRD File Format
+
+MRD data can be stored and streamed in various formats, including raw binary, HDF5, and NDJSON.
+
+We use [Yardl](https://microsoft.github.io/yardl/) to generate C++, Python, and MATLAB code for reading and writing MRD streams.
+
+## Binary
+
+The `mrd` binary format is the primary format for streaming raw MR data.
+
+See also: [Yardl Binary Encoding Reference](https://microsoft.github.io/yardl/reference/binary.html)
+
+Binary `mrd` streams can be read and written using C++, Python, and/or MATLAB.
+
+
+## HDF5
```

Comment:
I think we should remove this. HDF5 is only supported in C++ at the moment and I am not sure we want to keep that going forward.